content/docs/userspace/gall-guide/types.md

```diff
@@ -0,0 +1,221 @@
+
+You would typically use `!>` as part of a [`cage`](#cage) when you're
+constructing a `card` like a poke or a `%fact` gift to be sent off.
+
+### Extract data from `vase`
+
+The [zapgal](https://urbit.org/docs/hoon/reference/rune/zap#-zapgal) rune (`!<`)
+takes two arguments: A mold specifying the type to try and extract the data as,
+and the vase to be extracted.
+
+Let's look at an example in the dojo. First, let's create a vase of `[@t @ux @ud]`:
+
+```
+> =myvase !>(['foo' 0xabcd 123])
+> myvase
+[#t/[@t @ux @ud] q=[7.303.014 43.981 123]]
+```
+
+Next, let's try extracting our vase:
+
+```
+> !<  [@t @ux @ud]  myvase
+['foo' 0xabcd 123]
+```
+
+Now let's try asking for a `@p` rather than `@t`:
+
+```
+> !<  [@p @ux @ud]  myvase
+-need.@p
+-have.@t
+nest-fail
+```
+
+As you can see, it will crash if the type does not exactly match. Note that
+rather than using `!<`, you can also just clam the tail of the vase like:
+
+```
+> ((trel @t @ux @ud) +.myvase)
+[p='foo' q=0xabcd r=123]
+```
+
+The only problem is that you can't tell if the auras were wrong:
+
+```
+> ((trel @p @ud @ux) +.myvase)
+[p=~sibtel-tallyd q=43.981 r=0x7b]
+```
+
+You'd typically use `!<` on the data in `card`s that come in from other ships,
+agents, etc.
+
+## `mark`
+
+The `mark` type is just a `@tas` like `%foo`, and specifies the Clay filetype of
+some data. The `mark` corresponds to a mark file in the `/mar` directory, so a
+`mark` of `%foo` corresponds to `/mar/foo/hoon`. Mark files are used for saving
+data in Clay, validating data sent between agents or over the network, and
+converting between different data types. For more information about mark files,
+you can refer to the [Marks section of the Clay
+documentation](/docs/arvo/clay/marks/marks).
+
+## `cage`
+
+A `cage` is a cell of a [`mark`](#mark) and a [`vase`](#vase), like `[%foo !>('bar')]`. The data in the vase should match the data type of the specified
+mark.
+
+Most data sent between agents, either locally or over the network, will be in a
+cage. When data comes in, Gall will validate the data in the vase with the given
+mark. Additionally, the data may be converted to a different mark if appropriate
+conversion routines exist in the mark file. This is most commonly done when
+talking to a web front-end via Eyre's channel system - outgoing data will be
+automatically converted to a `%json` mark and incoming data will be converted
+from `%json` to the mark your agent expects.
+
+## `quip`
+
+`(quip a b)` is equivalent to `[(list a) b]`, it's just a more convenient way to
+specify it. Most arms of an agent return a `(quip card _this)`, which is a list
+of effects and a new state.
+
+## `path`
+
+The `path` type is formally defined as:
+
+```hoon
++$  path  (list knot)
+```
+
+A knot is a `@ta` text atom (see the [Strings guide](/docs/hoon/guides/strings)
+for details), so a `path` is just a list of text. Rather than having to write
+`[~.foo ~.bar ~.baz ~]` though, it has its own syntax which looks like
```

Comment:
For whatever reason, the pretty printer puts out
```
> /foo/bar/baz
[%foo %bar %baz ~]
```
instead of `[~.foo ~.bar ~.baz ~]`. I guess this may just be a bug? Not sure if its worth mentioning here or not, but I know I've been confused by this before.